dandiapi/api/tasks/__init__.py

```diff
@@ -184,3 +185,45 @@ def unembargo_dandiset(dandiset_id: int):
     # Set access on dandiset
     dandiset.embargo_status = Dandiset.EmbargoStatus.OPEN
     dandiset.save(update_fields=['embargo_status'])
+
+
+@shared_task
+@atomic
+def publish_task(version_id: int):
+    old_version: Version = Version.objects.get(id=version_id)
+    new_version: Version = old_version.publish_version
+
+    new_version.doi = doi.create_doi(new_version)
```

Comment:
This is unrelated, but could you make a follow on PR moving this into on_commit? This is a bug waiting to happen.

Reply:
Could you explain further about the bug risk?

Reply:
@waxlamp as it is now, if something goes wrong and the transaction gets rolled back, the DOI is still created.

Reply:
Ah makes sense, thanks!